pocs/utils/pyro/__init__.py

```diff
@@ -0,0 +1,148 @@
+            addresses = [address for address in addresses if address.split('.')[1] == byte2]
+
+    assert len(addresses) == 1
+    host = addresses[0]
+    msg = 'Using IP address {} on interface {}'.format(addresses[0], interface)
+    if verbose:
+        print(msg)
+    if logger:
+        logger.debug(msg)
+    return host
+
+
+def run_name_server(host=None, port=None, autoclean=0):
+    """
+    Runs a Pyro name server.
+
+    The name server must be running in order to use distributed cameras with POCS. The name server
+    should be started before starting camera servers or POCS.
+
+    Args:
+        host (str, optional): hostname/IP address to bind the name server to. If not given then
+            get_own_ip will be used to attempt to automatically determine the IP addresses of
+            the computer that the name server is being started on.
+        port (int, optional): port number to bind the name server to. If not given then the port
+            will be selected automatically (usually 9090).
+        autoclean (int, optional): interval, in seconds, for automatic deregistration of objects
+            from the name server if they cannot be connected. If not given no autocleaning will
+            be done.
+    """
+    try:
+        # Check that there isn't a name server already running
+        name_server = Pyro4.locateNS()
+    except errors.NamingError:
+        if not host:
+            # Not given an hostname or IP address. Will attempt to work it out.
+            host = get_own_ip(verbose=True)
+        else:
+            host = str(host)
+
+        if port:
+            port = int(port)
+        else:
+            port = None
+
+        Pyro4.config.NS_AUTOCLEAN = float(autoclean)
+
+        print("Starting Pyro name server... (Control-C/Command-C to exit)")
+        naming.startNSloop(host=host, port=port)
+    else:
+        print("Pyro name server {} already running! Exiting...".format(name_server))
+
+
+def run_camera_server(ignore_local):
+    """
+    Runs a Pyro camera server.
+
+    The camera server should be run on the camera control computers of distributed cameras. The
+    camera servers should be started after the name server, but before POCS.
+
+    Args:
+        ignore_local (bool, optional): If True use the default $POCS/conf_files/pyro_camera.yaml
+            only. If False will allow $POCS/conf_files/pyro_camera_local.yaml to override the
+            default configuration. Default False.
+    """
+    Pyro4.config.SERVERTYPE = "multiplex"
+    config = load_config(config_files=['pyro_camera.yaml'], ignore_local=ignore_local)
+    host = config.get('host', None)
+    if not host:
+        host = get_own_ip(verbose=True)
+    port = config.get('port', 0)
+
+    with Pyro4.Daemon(host=host, port=port) as daemon:
+        try:
+            name_server = Pyro4.locateNS()
+        except errors.NamingError as err:
+            warn('Failed to locate Pyro name server: {}'.format(err))
```

Comment:
Need to import `warn`

Reply:
Oops. Need to watch out for that sort of thing when cutting & pasting functions between files!